defaults/stroopwafel_interface_default.py

```diff
@@ -0,0 +1,286 @@
+    return [m1, q, a]
+
+def update_properties(locations, dimensions):
+    """
+    This function is not mandatory, it is required only if you have some dependent variable. 
+    For example, if you want to sample Mass_1 and q, then Mass_2 is a dependent variable which is product of the two.
+    Similarly, you can assume that Metallicity_2 will always be equal to Metallicity_1
+    IN:
+        locations (list(Location)) : A list containing objects of Location class in classes.py. 
+        You can play with them and update whatever fields you like or add more in the property (which is a dictionary)
+    OUT: Not Required
+    """
+    m1 = dimensions[0]
+    q = dimensions[1]
+    for location in locations:
+        location.properties['Mass_2'] = location.dimensions[m1] * location.dimensions[q]
+        location.properties['Metallicity_2'] = location.properties['Metallicity_1'] = constants.METALLICITY_SOL
+        location.properties['Eccentricity'] = 0
+        #location.properties['Kick_Mean_Anomaly_1'] = np.random.uniform(0, 2 * np.pi, 1)[0]
+        #location.properties['Kick_Mean_Anomaly_2'] = np.random.uniform(0, 2 * np.pi, 1)[0]
+
+
+
+
+
+#################################################################################
+#################################################################################
+###                                                                           ###
+###         USER SHOULD NOT SET ANYTHING BELOW THIS LINE                      ###
+###                                                                           ###
+#################################################################################
+#################################################################################
+
+
+
+
+
+def configure_code_run(batch):
+    """
+    This function tells stroopwafel what program to run, along with its arguments.
+    IN:
+        batch(dict): This is a dictionary which stores some information about one of the runs. It has an number key which stores the unique id of the run
+            It also has a subprocess which will run under the key process. Rest, it depends on the user. User is free to store any information they might need later 
+            for each batch run in this dictionary. For example, here I have stored the 'output_container' and 'grid_filename' so that I can read them during discovery of interesting systems below
+    OUT:
+        compas_args (list(String)) : This defines what will run. It should point to the executable file along with the arguments.
+        Additionally one must also store the grid_filename in the batch so that the grid file is created
+    """
+    batch_num = batch['number']
+    grid_filename = os.path.join(output_folder, 'grid_' + str(batch_num) + '.csv')
+    output_container = 'batch_' + str(batch_num)
+    compas_args = [compas_executable, '--grid', grid_filename, '--output-container', output_container, '--random-seed' , np.random.randint(2, 2**63 - 1)]
```

Comment:
It doesn't quite work to leave the random seed in from the pythonSubmit, because the same seed is passed to each new batch (i.e for each call of COMPAS). It's not a huge problem, since the grids should all be different (which covers most of what was previously handled by the random number), but any numbers that are pulled purely from a random draw will be repeated. 

A possible fix to this would be use the random-seed from pythonSubmit, but incremented by the number of binaries in a given batch times the index of the batch. I'll give that a go and see if I run into any problems.

Reply:
this has now been fixed